src/Codeception/Lib/InnerBrowser.php

```diff
@@ -439,29 +439,47 @@ protected function clickByLocator($link)
      */
     private function clickButton(\DOMNode $node)
     {
-        $formParams = [];
-        $buttonName = (string)$node->getAttribute('name');
-        $buttonValue = $node->getAttribute('value');
-
-        if ($buttonName !== '' && $buttonValue !== null) {
-            $formParams = [$buttonName => $buttonValue];
+        /**
+         * First we check if the button is associated to a form.
+         * It is associated to a form when it has a nonempty form
+         */
+        $formAttribute = $node->attributes->getNamedItem('form');
+        if (isset($formAttribute)) {
+            $form = empty($formAttribute->nodeValue) ? null : $this->filterByCSS('#' . $formAttribute->nodeValue)->getNode(0);
```

Comment:
`if` reads better than ternary.
It would be better to move `$form = null;` from `else` to line above `if` and change this line to
```php
if (!empty($formAttribute->nodeValue)) {
    $form = $this->filterByCSS('#' . $formAttribute->nodeValue)->getNode(0);
}

Reply:
Actually that's not the same.
If the form attribute exists, we MUST base our value for `$form` on it; even if it is empty.

So the first guard makes that distinction
```php
if (isset($formAttribute)) {
    // Form attribute exists
    ...
        } else {
    // Form attribute does not exist
    ...
}

```
The ternary operator clearly indicates that we are doing an assignment in both paths, an if statement could do anything and is therefore, in my opinion, more expensive from a cognitive point of view.